src/NServiceBus.RavenDB.Tests/Outbox/When_adding_outbox_messages.cs

```diff
@@ -0,0 +1,139 @@
+﻿namespace NServiceBus.RavenDB.Tests.Outbox
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using NServiceBus.Outbox;
+    using NServiceBus.RavenDB.Outbox;
+    using NUnit.Framework;
+    using Raven.Abstractions.Exceptions;
+    using Raven.Client.Exceptions;
+
+    [TestFixture]
+    public class When_adding_outbox_messages : RavenTestBase
+    {
+        [SetUp]
+        public override void SetUp()
+        {
+            base.SetUp();
+            new OutboxRecordsIndex().Execute(store);
+        }
+
+        [Test]
+        [ExpectedException(typeof(NonUniqueObjectException))]
+        public void Should_throw_if__trying_to_insert_same_messageid()
+        {
+            var sessionFactory = new RavenSessionFactory(store);
+            var persister = new OutboxPersister(sessionFactory);
+
+            using (sessionFactory.Session)
+            {
+                persister.Store("MySpecialId", Enumerable.Empty<TransportOperation>());
+                persister.Store("MySpecialId", Enumerable.Empty<TransportOperation>());
+
+                sessionFactory.SaveChanges();
+            }
+        }
+
+        [Test]
+        [ExpectedException(typeof(ConcurrencyException))]
```

Comment:
I know we have these all through the code base but `ExpectedException` attributes are basically an anti pattern. The main reason being you have no control over what line you are expecting the exception to happen. Recommend using `Assert.Throws<T>` instead